### DISCUSSION TITLE
cs/system/pa/

### DISCUSSION THREAD

**Discussion by giscus** [TonyCrane/note#23] [General]
# cs/system/pa/

鹤翔万里（TonyCrane）的笔记本

https://note.tonycrane.cc/cs/system/pa/

**Comment by phoulx** [comment]
Linux改键推荐用keyd ( https://github.com/rvaiya/keyd )，基本可以复刻Mac键位，或者任何键位偏好。

**Comment by LQ0623** [comment]
你好，我想请问一下在PA2.3的时候，如果开启了diff test，运行外设相关的操作就会报寄存器值不一样，这个问题请问您解决了吗？

**Comment by sigerio** [comment]
大佬你好，我参考https://github.com/ics-nju-wl/icspa-public-guide/blob/master/ch/ch_pa-1-1_reg.md这里的x86实验指导，然后在PA1里面打印寄存器部分，通过printf("%s %x %08x\n",regs[i],0,cpu.gpr[i]);代码来进行实现，但是实际打印cpu.gpr[i]的值全是0，请问是不是我的代码不对

**Reply by sigerio** [reply]
补充一点，我选的是risc-v32的系统

**Reply by sigerio** [reply]
大佬的意思是，先看手册？我不知道自己方向对不对，大佬给指点下

**Comment by mmjwxbc** [comment]
大佬，我想问一下pa2.3需要在device目录下实现什么代码吗，我运行hello会往控制台输出，反倒是运行alutest，啥输出都没有
